src/renderer/runner.ts

```diff
@@ -387,31 +387,44 @@ export class Runner {
     // Add user-specified cli flags if any have been set.
     const options = [dir, '--inspect'].concat(executionFlags);
 
+    const cleanup = async () => {
+      flushOutput();
+
+      this.appState.isRunning = false;
+      this.child = null;
+
+      // Clean older folders
+      await window.ElectronFiddle.app.fileManager.cleanup(dir);
+      await this.deleteUserData();
+    };
+
     return new Promise(async (resolve, _reject) => {
-      this.child = await fiddleRunner.spawn(
-        isValidBuild && localPath ? Installer.getExecPath(localPath) : version,
-        dir,
-        { args: options, cwd: dir, env },
-      );
+      try {
+        this.child = await fiddleRunner.spawn(
+          isValidBuild && localPath
+            ? Installer.getExecPath(localPath)
+            : version,
+          dir,
+          { args: options, cwd: dir, env },
+        );
+      } catch (e) {
+        pushOutput(`Failed to spawn Fiddle: ${e.message}`);
+        await cleanup();
+        resolve(RunResult.FAILURE);
+      }
+
       this.appState.isRunning = true;
```

Comment:
Even if the above code triggers a catch, it will still be executed here. So that's what I'm talking about: we need `return`